out.rs

```diff
@@ -0,0 +1,1089 @@
+                type Sign;
+            }
+
+            mod dec2flt {
+                mod algorithm {
+                    mod fpu_precision {
+                        type FPUControlWord;
+                    }
+
+                }
+
+                mod rawfp {
+                    type Unpacked;
+                    trait RawFloat {
+                        fn classify(self) -> FpCategory;
+                        fn integer_decode(self) -> ( u64 , i16 , i8 );
+                        fn unpack(self) -> Unpacked;
+                    }
+                }
+
+                mod parse {
+                    type Sign;
+                    type Decimal;
+                    type ParseResult;
+                }
+
+                type ParseFloatError;
+                type FloatErrorKind;
+                impl ParseFloatError {
+                    fn __description(& self) -> & str;
+                }
+            }
+
+            mod bignum {
+                trait FullOps {
+                }
+                type Digit32;
+            }
+
+            mod diy_float {
+                type Fp;
+                impl Fp {
+                    fn mul(& self, & Fp) -> Fp;
+                    fn normalize(& self) -> Fp;
+                    fn normalize_to(& self, i16) -> Fp;
+                }
+            }
+
+            impl u8 {
+                fn is_ascii(& self) -> bool;
+                fn to_ascii_uppercase(& self) -> u8;
+                fn to_ascii_lowercase(& self) -> u8;
+                fn eq_ignore_ascii_case(& self, & u8) -> bool;
+                fn make_ascii_uppercase(& mut self) ;
+                fn make_ascii_lowercase(& mut self) ;
+                fn is_ascii_alphabetic(& self) -> bool;
+                fn is_ascii_uppercase(& self) -> bool;
+                fn is_ascii_lowercase(& self) -> bool;
+                fn is_ascii_alphanumeric(& self) -> bool;
+                fn is_ascii_digit(& self) -> bool;
+                fn is_ascii_hexdigit(& self) -> bool;
+                fn is_ascii_punctuation(& self) -> bool;
+                fn is_ascii_graphic(& self) -> bool;
+                fn is_ascii_whitespace(& self) -> bool;
+                fn is_ascii_control(& self) -> bool;
+            }
+            type FpCategory;
+            type TryFromIntError;
+            impl TryFromIntError {
+                fn __description(& self) -> & str;
+            }
+            trait FromStrRadixHelper {
+            }
+            type ParseIntError;
+            type IntErrorKind;
+            impl ParseIntError {
+                fn kind(& self) -> & IntErrorKind;
+                fn __description(& self) -> & str;
+            }
+        }
+
+        mod mem {
+            mod manually_drop {
+                type ManuallyDrop;
+            }
+
+            type Discriminant;
+        }
+
+        mod ptr {
+            mod non_null {
+                type NonNull;
+            }
+
+            mod unique {
```

Comment:
I see a lot of things in this file that are not public APIs. For example `unique` is not a module in `std::ptr` -- https://doc.rust-lang.org/1.38.0/std/ptr/index.html -- and should not be exposed in our standard library definition. If it isn't possible to reliably programmatically identify what parts of the standard library are public, then this is the wrong approach and we should not be trying to generate the whole thing this way.

Reply:
Yeah the code actually is not verifying the visibility of items. Will fix that.